methods/inputs/locate_gedi_data.py

```diff
@@ -103,48 +124,106 @@ def gedi_fetch(boundary_file: str, gedi_data_dir: str) -> None:
             new_feature = ogr.Feature(feature_definition)
             new_feature.SetGeometry(geometry)
             working_layer.CreateFeature(new_feature)
-            del destination_data_source # aka destination_data_source.Close()
+            del destination_data_source  # close the GeoJSON file
 
+            # read the GeoJSON as a GeoPandas dataframe for further processing
             shape = gpd.read_file(chunk_path)
 
-            # We set max_coords rather low (the upper bound is less than 5000) as some of
-            # our geometries are causing issues with being too precise and the coordinates
-            # not all being > 1m apart. Checking this for every geometry is a bit hard, so
-            # instead we approximate it by setting the bar for simplification much lower.
+            # simplify the shape to avoid excessive precision issues in coordinates
             shape = check_and_format_shape(shape, simplify=True, max_coords=200)
+
+            # query the GEDI data for this shape and specified time range
             result = query(
                 product=GediProduct.L4A,
-                date_range=(dt.datetime(2020, 1, 1, 0, 0), dt.datetime(2021, 1, 1, 0, 0)),
+                date_range=(dt.datetime(start_year, 1, 1), dt.datetime(end_year, 1, 1)),
                 spatial=shape
             )
             granule_metadatas.append(result)
 
+            # move to the next chunk
             chunk = chunked_layer.GetNextFeature()
 
+    # concatenate all metadata into a single dataframe and remove duplicates
     granule_metadata = pd.concat(granule_metadatas).drop_duplicates(subset="granule_name")
 
+    # write the granule names to a text file in the specified directory
+    granule_names = granule_metadata["granule_name"].tolist()
+    output_txt_path = os.path.join(output_txt_dir, "gedi_names.txt")
```

Comment:
In general I prefer it if the output name is an argument, so when you read the pipeline script that Eping and Abby wrote you can see where each input and output comes from - reading the script it might not be clear where "gedi_names.txt" comes from. Not essential, just a preference from me.

Reply:
Yes that's a good idea, I'll see if I can make it so the name is the same as the name of the GeoJson but "{GeoJson Name}-gedi_names.csv"